test/test_segment_reductions.py

```diff
@@ -83,21 +85,29 @@ def _test_common(
                 )
             )
 
-    @dtypesIfCUDA(torch.half, torch.bfloat16, torch.float, torch.double)
     @dtypes(torch.half, torch.bfloat16, torch.float, torch.double)
     def test_simple_1d(self, device, dtype):
         lengths = [1, 2, 3, 0]
         data = [1, float("nan"), 3, 4, 5, 5]
-        initial_value = 0
         check_backward = True
 
-        for reduction in ("max", "mean"):
+        for reduction in reductions:
```

Comment:
it looks like you are only testing with defined initial_value, would be good to test with undefined also. Unfortunately, some cuda cases will break in unsafe mode, so you need to avoid that.